From b8b702e25f3602223102c238ae3228d28ce9b2d5 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Fri, 5 Sep 2014 02:33:30 -0400
Subject: [iommu] vt-d: Use correct domain id to flush virtual machine domains

Message-id: <20140905023330.497.54877.stgit@gir.stowe>
Patchwork-id: 91384
O-Subject: [RHEL7.1 PATCH 7/7] iommu/vt-d: Use correct domain id to flush virtual machine domains
Bugzilla: 1136539
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Don Dutile <ddutile@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1136539
Upstream Status: 18fd779a41eff52b597bd2d327194f98a83de4af

commit 18fd779a41eff52b597bd2d327194f98a83de4af
Author: Jiang Liu <jiang.liu@linux.intel.com>
Date:   Fri Jul 11 14:19:26 2014 +0800

    iommu/vt-d: Use correct domain id to flush virtual machine domains

    For virtual machine domains, domain->id is a virtual id, and the real
    domain id written into context entry is dynamically allocated.
    So use the real domain id instead of domain->id when flushing iotlbs
    for virtual machine domains.

    Signed-off-by: Jiang Liu <jiang.liu@linux.intel.com>
    Signed-off-by: Joerg Roedel <jroedel@suse.de>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/intel-iommu.c b/drivers/iommu/intel-iommu.c
index 689ea4b..4bf8989 100644
--- a/drivers/iommu/intel-iommu.c
+++ b/drivers/iommu/intel-iommu.c
@@ -1822,7 +1822,7 @@ static int domain_context_mapping_one(struct dmar_domain *domain,
         (((u16)bus) << 8) | devfn,
         DMA_CCMD_MASK_NOBIT,
         DMA_CCMD_DEVICE_INVL);
-  iommu->flush.flush_iotlb(iommu, domain->id, 0, 0, DMA_TLB_DSI_FLUSH);
+  iommu->flush.flush_iotlb(iommu, id, 0, 0, DMA_TLB_DSI_FLUSH);
  } else {
   iommu_flush_write_buffer(iommu);
  }
-- 
1.7.1